shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -345,12 +345,18 @@ public long performNativeAttach(@NonNull FlutterJNI flutterJNI) {
   public FlutterJNI spawn(
       @Nullable String entrypointFunctionName,
       @Nullable String pathToEntrypointFunction,
-      @Nullable String initialRoute) {
+      @Nullable String initialRoute,
+      @Nullable List<String> entrypointArgs) {
     ensureRunningOnMainThread();
     ensureAttachedToNative();
+    String[] args = entrypointArgs != null ? entrypointArgs.toArray(new String[0]) : new String[0];
```

Comment:
I'm not a fan of introducing an extra copy of the arguments.  Just send a List and read from in in JNI.

Reply:
Done